### PR TITLE
validate sample and patient before render the patient view

### PR DESCRIPTION
**Pull request by dippindots**
validate sample and patient before render the patient view
Fix # https://github.com/cbioportal/cbioportal/issues/5357.

Changes proposed in this pull request:
- a show error when sample or patient not exists
image

### CODE REVIEW

**Comment by dippindots** [comment]
test links:
1. [sample](https://cbioportal-frontend-pr-2728.herokuapp.com/patient?sampleId=P-0035148-T01-IM6&studyId=msk_impact_2017)
2. [patient](https://cbioportal-frontend-pr-2728.herokuapp.com/patient?studyId=ucec_tcga_pub&caseId=not-a-sample)

**Comment by alisman** [comment]
@dippindots 
For this case, we should not say "URL invalid."  The URL form is valid, it's just that the ID doesn't exist.   So, error message should be that Patient doesn't exist.

File: src/pages/patientView/PatientViewPage.tsx [hunk 1]
```diff
@@ -109,12 +109,20 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
...
-                    }
-                    if ('caseId' in query) {
-                        patientViewPageStore.setPatientId(query.caseId as string);
-                    } else if ('sampleId' in query)
-                    {
-                        patientViewPageStore.setSampleId(query.sampleId as string);
+
+                        if ('caseId' in query) {
+                            patientViewPageStore.setPatientId(query.caseId as string);
+                            const validationPatient = await validatePatient(query.caseId, query.studyId);
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx]: I think this check belongs in the patientViewStore, not here.

**Comment by alisman** [comment]
When a user submits a request with invalid sample/patient, we already get 404s.  So we can accomplish this ticket by properly handling those errors states for the existing mobx promises.  In other words, we should just show a message when we detect isError=true

here is example 
https://www.cbioportal.org/patient?studyId=ucec_tcga_pub&caseId=TCG-BK-A0CC

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 2]
```diff
@@ -267,6 +269,9 @@ export class PatientViewPageStore {
     readonly samples = remoteData(
         {
             invoke: async () => fetchSamplesForPatient(this.studyId, this._patientId, this.sampleId),
+            onError: (err: Error) => {
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: per discussion, test for specific error situations and then add via appStore

File: src/shared/components/errorContainer/ErrorContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
```
Comment (alisman) [inline src/shared/components/errorContainer/ErrorContainer.tsx]: this will then be unnecessary

Comment (dippindots) [reply src/shared/components/errorContainer/ErrorContainer.tsx]: Yes, I will use ErrorScreen rather than this.

**Pull request merged**